contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -352,16 +380,39 @@ contract LidoOracle is ILidoOracle, AragonApp {
             .setStorageUint256(_allowedBeaconBalanceRelativeDecrease);
         emit AllowedBeaconBalanceRelativeDecreaseSet(_allowedBeaconBalanceRelativeDecrease);
 
-        // set last completed epoch as V1's contract last reported epoch, in the vast majority of
-        // cases this is true, in others the error is within a frame
-        uint256 lastReportedEpoch = V1_LAST_REPORTED_EPOCH_ID_POSITION.getStorageUint256();
-        LAST_COMPLETED_EPOCH_ID_POSITION.setStorageUint256(lastReportedEpoch);
-
-        // set expected epoch to the first epoch for the next frame
+        // // set expected epoch to the first epoch for the next frame
         BeaconSpec memory beaconSpec = _getBeaconSpec();
-        uint256 expectedEpoch = _getFrameFirstEpochId(lastReportedEpoch, beaconSpec) + beaconSpec.epochsPerFrame;
+        uint256 expectedEpoch = _getFrameFirstEpochId(0, beaconSpec) + beaconSpec.epochsPerFrame;
         EXPECTED_EPOCH_ID_POSITION.setStorageUint256(expectedEpoch);
         emit ExpectedEpochIdUpdated(expectedEpoch);
+
+        // Initializations for v2 --> v3
+        _initialize_v3();
+
+        // Need this despite contract version check because Aragon requires it to handle auth() modificators properly
+        initialized();
+    }
+
+
+    /**
+     * @notice A function to finalize upgrade to v3 (from v1). Can be called only once
```

Comment:
It's not clear (from the comment context) why we allow upgrading from v1 to v3 and not from v2 to v3.